src/core/tags_parser.hpp

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include "parse_feeder.hpp"
+#include "parser.hpp"
+
+namespace docgen {
+namespace core {
+
+struct TagParser_routines
+{
+	protected:
+		static constexpr const routine_t on_tag_  = [](const token_t& t) {
+			ParseFeeder::at(t.content);
+			ParseFeeder::go();
+			ParseFeeder::skip();
+		};
+};
+
+class TagParser : public ParseWorker, private TagParser_routines
```

Comment:
To be more consistent with the name, I think `TagParseWorker` is more intuitive. See my comments about `Parser` - I don't think it should be a `ParseWorker`. It makes sense, however, for this class to derive from `ParseWorker`, hence the proposed name change.

Reply:
This was my original idea, but something about the three word camel uppercase doesn't sit right with me :(

Reply:
I prefer consistency over style any day (of course consistent good style is the best).